daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SError.scala

```diff
@@ -67,7 +67,7 @@ object SError {
   final case class DamlELocalContractNotActive(
       coid: ContractId,
       templateId: TypeConName,
-      consumedBy: NodeId,
+      consumedBy: Option[PartialTransaction.Tree],
```

Comment:
The Option here is a bit ugly.
I think the info carries by the domain of the map was before not very useful.
In fact I am pretty sure this is useful only in the pretty printer to "explain" why a transaction failed.

I believe the Map can be replaced by a Set
```suggestion
      consumed: Set[Value.ContractId],
```
the you can recompute which node consumed the contract directly from the partially transaction when you need it (you just look for the first consuming node that exercised the contract)